sound/pci/hda/patch_hdmi.c

```diff
@@ -2654,7 +2654,7 @@ static void generic_acomp_pin_eld_notify(void *audio_ptr, int port, int dev_id)
 	/* skip notification during system suspend (but not in runtime PM);
 	 * the state will be updated at resume
 	 */
-	if (snd_power_get_state(codec->card) != SNDRV_CTL_POWER_D0)
+	if (codec->core.dev.power.power_state.event == PM_EVENT_SUSPEND)
```

Comment:
this looks to me like too deep an intrusion into PM object internals, into its presumably private fields, for which there should be higher-level accessors, but I don't see any such accessors available and looks like other drivers perform the same kind of access too...

Reply:
@lyakh Yeah, I was wondering the same, but this is very commonplace in "surrounding code", both in codec and controller drivers, so I think this is ok.